### PR TITLE
Feat/sirh 89 endpoint to update password

### PR DESCRIPTION
**Pull request by alexeh**
Resets a user's password providing a previously generated token and a new password

### CODE REVIEW

File: shared/contracts/user.contract.ts [hunk 1]
```diff
@@ -102,4 +103,13 @@ export const userContract = contract.router({
       400: contract.type<JSONAPIError>(),
     },
   },
+  resetPassword: {
+    method: 'POST',
+    path: '/users/me/password/reset',
+    responses: {
+      200: contract.type<ApiResponse<UserDto>>(),
```
Comment (alexeh) [inline shared/contracts/user.contract.ts:110]: Do we really need to return the user (with no sessions) after updating the password?

Should we internally generate a session and return also a new access token?

Besides that, when a token is generated and sent through the email, this is signed with the same secret for the access token even tho it has a different expiring date.

Should we sign the recovery password token with a different secret?

File: shared/contracts/user.contract.ts [hunk 2]
```diff
@@ -102,4 +103,13 @@ export const userContract = contract.router({
       400: contract.type<JSONAPIError>(),
     },
   },
+  resetPassword: {
+    method: 'POST',
+    path: '/users/me/password/reset',
```
Comment (alexeh) [inline shared/contracts/user.contract.ts:108]: following the approach of moving the recover password flow to the auth controller given there is "no user" at that point, Ive decided to put this in within the user module given that the user has to present a token tied to its id, so in this case "there is a user"

Comment (andresgnlez) [reply shared/contracts/user.contract.ts:108]: LGTM. At this point we are already dealing with the user so makes sense to me keeping the method here.

File: shared/contracts/user.contract.ts (hunk 1 above)
Comment (andresgnlez) [reply shared/contracts/user.contract.ts:110]: No, we should only return the user in two cases: 
- the user has logged in and we retrieve their info for convenience
- the user is triggering requests where their token is attached so we know in advanced who they are.

In the special case of the recovery password, which, yeah, has an exclusive token for the only purpose of knowing what user we are updating the password for I would stick with a final confirmation (200) for the password change and encourage the user to go to the signin page and log in as usual with their new password.

**Pull request merged**